posts/2020-08-16-ocp-log-forwarding.adoc

```diff
@@ -0,0 +1,390 @@
+...
+<label @CONTAINER_LOGS>
+  <match **>
+    @type copy
+
+    <store>
+      @type relabel
+      @label @ELASTICSEARCH
+    </store>
+    <store>
+      @type relabel
+      @label @ELASTICSEARCH-INSECURE
+    </store>
+    <store>
+      @type relabel
+      @label @FLUENTD-FORWARD
+    </store>
+  </match>
+</label>
+...
+```
++
+
+
+== Internal and External Splunk configuration
+
+Using `forward` output, you can forward OpenShift Container Platform logs to Splunk using Fluentd forward protocol. You can setup Splunk inside your OpenShift Cluster or on your external machine.
+
+=== Option 1: Deploying Splunk onto your OpenShift cluster
+
+This option deploys an instance of Splunk inside your OpenShift cluster using a script. If you do not have Splunk deployed already, this option will make Splunk setup simple through a usage of pre-created configurations.
+
+. Download and Install the following tools:
+* Git
+* OpenShift Command Line Tool
+* Helm Command Line Tool
+
+. Login to your OpenShift through command line tool with a user with `cluster-admin` permissions.
+
+. Clone the git repository to your machine. The repository was retrieved from OpenShift blog post: link:https://www.openshift.com/blog/forwarding-logs-to-splunk-using-the-openshift-log-forwarding-api[Forwarding Logs to Splunk Using the OpenShift Log Forwarding API].
++
+[source]
+----
+[root@ocp ~]# git clone https://github.com/sabre1041/openshift-logforwarding-splunk.git
+[root@ocp ~]# cd openshift-logforwarding-splunk
+----
++
+
+. Deploy a nonpersistent instance of splunk to a project called `splunk`.
++
+[source]
+----
+[root@ocp ~]# ./splunk-install.sh
+----
++
+Once splunk is deployed, you can login by discovering the exposed link.
++
+[source]
+----
+[root@ocp ~]# echo https://$(oc get routes -n splunk splunk -o jsonpath='{.spec.host}')
+----
++
+The default credentials are:
++
+```
+Username: admin
+Password: admin123
+```
++
+
+. Retrieve token value in Splunk's Settings > Data Inputs > HTTP Event Collector. Install the Helm chart to the OpenShift environment. Helm chart 
+deploys the Fluentd forwarder and OpneShift Log Forwarding API components. Replace _<token>_ with token that you retrieved from Splunk.
++
+[source]
+----
+[root@ocp ~]# helm upgrade -i --wnamespace=openshift-logging openshift-logforwarding-splunk charts/openshift-logforwarding-splunk/ --set forwarding.splunk.token=<token>
+----
++
+Confirm *openshift-logforwarding-splunk* pods are running.
++
+[source]
+----
+[root@ocp ~]# oc get pods -n openshift-logging
+----
++
+
+=== Option 2: Setting up Splunk on your external machine
+
+This option sets up Splunk manually on your external machine. If you already have Splunk deployed on your external machine, this option will help you setup the connection between your OpenShift cluster and Splunk. You can deploy Fluentd and Splunk using docker compose. This way, obtaining docker images, setting up configurations and running the applications can be done easily.
```

Comment:
should mention here that this docker compose way of doing setup is really for demo purposes

Reply:
also would be good to educate reader on the fact that Fluentd forward forwards to another Fluentd instance, which is why you need it in front of your external Splunk